### PR TITLE
refactor backend salt utils and added ceph cluster setup

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

File: usm_wrappers/salt_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+    gluster_minion = utils.resolve_ip_address(gluster_node)
+    new_minion = utils.resolve_ip_address(new_node)
+    out = local.cmd(gluster_minion, 'glusterfs.peer', [new_minion])
+    if out and 'success' in out[gluster_minion]:
+        return True
+    else:
+        return False
+
+
+import ConfigParser
```
Comment (ChristinaMeno) [inline usm_wrappers/salt_wrapper.py:151]: This looks like two files concatenated. Should they be separate?

File: usm_wrappers/salt_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+    pillar_data = _add_ceph_mon_pillar_data(mon_id_map, cluster_name, monitors)
+    pillar = {'usm': pillar_data}
+
+    bootstrapped_minion = None
+    for id, minion in mon_id_map.iteritems():
+        out = run_state(local, minion, 'add_ceph_mon',
+                        kwarg={'pillar':
+                               {'usm': {'mon_bootstrap': True,
+                                        minion: pillar_data[minion]}}})
+        if not out:
```
Comment (ChristinaMeno) [inline usm_wrappers/salt_wrapper.py:304]: So we fail if any mon fails to create?

File: usm_wrappers/salt_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+        ## mon failed to start
+        return False
+
+    minion_set = set(minions)
+    minion_set.remove(bootstrapped_minion)
+    if minion_set:
+        rv = run_state(local, minion_set, 'add_ceph_mon', expr_form='list',
+                       kwarg={'pillar': pillar})
+        if rv:
+            raise Exception('add_ceph_mon state failed somewhere')
```
Comment (ChristinaMeno) [inline usm_wrappers/salt_wrapper.py:329]: This is almost useless. Why not define an exception class like BootstrapException and say at least that we failed add_ceph_mon of a specific host

File: usm_wrappers/salt_wrapper.py [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+                failed_results.update({id: res})
+        if not v:
+            failed_minions[minion] = {}
+        if failed_results:
+            failed_minions[minion] = failed_results
+
+    return failed_minions
+
+
+def run_state(local, tgt, state, *args, **kwargs):
```
Comment (ChristinaMeno) [inline usm_wrappers/salt_wrapper.py:30]: We should be exposing more detail on failure here

File: usm_wrappers/utils.py [hunk 5]
```diff
@@ -0,0 +1,513 @@
...
+        raise CmdExecFailed(command, p.returncode, out, err)
+
+    if not raw:
+        out = out.splitlines(False)
+        err = err.splitlines(False)
+
+    return (p.returncode, out, err)
+
+
+paramiko.util.get_logger('paramiko').setLevel(logging.ERROR)
```
Comment (ChristinaMeno) [inline usm_wrappers/utils.py:409]: Why is this buried here?

File: usm_wrappers/salt_wrapper.py (hunk 1 above)
Comment (balamurugana) [inline usm_wrappers/salt_wrapper.py:151]: Yes. They should be separate and will be done in awhile

File: usm_wrappers/salt_wrapper.py (hunk 2 above)
Comment (balamurugana) [inline usm_wrappers/salt_wrapper.py:304]: We are going to change this logic ie setup_ceph_cluster() will expect only one mon and usm core will call add_ceph_mon() for rest of mons to have more control on error handling

File: usm_wrappers/salt_wrapper.py (hunk 3 above)
Comment (balamurugana) [inline usm_wrappers/salt_wrapper.py:329]: Yes.  It will be done that way.

File: usm_wrappers/salt_wrapper.py (hunk 4 above)
Comment (balamurugana) [inline usm_wrappers/salt_wrapper.py:30]: Yes.  The error handling is not there and will be fixed soon.

File: usm_wrappers/utils.py (hunk 5 above)
Comment (balamurugana) [inline usm_wrappers/utils.py:409]: Basically we would need to set log level of paramiko.  I felt the util module should take care of it as api consumer doesn't care about paramiko whereas in rexecCmd().  Do you think this has to be in consumer level?

**Comment by ChristinaMeno** [comment]
looks fine

**Pull request merged**